flake.nix

```diff
@@ -31,50 +31,78 @@
           extensions = ["rust-src" "rust-analyzer"];
         };
 
-        runtimeDeps = with pkgs;
-          if stdenv.isDarwin
-          then [
-            darwin.libobjc
-            darwin.apple_sdk_11_0.frameworks.AppKit
-            darwin.apple_sdk_11_0.frameworks.AVFoundation
-            darwin.apple_sdk_11_0.frameworks.Vision
-          ]
-          else
-            (with pkgs; [
-              (lib.getLib gcc-unwrapped)
-              fontconfig
-              libGL
-              libxkbcommon
-              vulkan-loader
-              wayland
-            ])
-            ++ (with pkgs.xorg; [
-              libX11
-              libXcursor
-              libXi
-              libXrandr
-              libxcb
-            ]);
-
-        buildDeps = with pkgs;
-          [
-            ncurses
-          ]
-          ++ lib.optionals stdenv.isLinux [
-            pkg-config
-            cmake
-            autoPatchelfHook
-          ];
-
-        rustPackage = rust-toolchain:
-          pkgs.rustPlatform.buildRustPackage {
+        mkRio = {
```

Comment:
This seems like an old snippet from old config? What was the motivation for this? However, I just tested this change and seems working on this end. Is this change necessary? I believe that we should not be adding additional functions on functions just for the dependencies. We could add these for `PerSystem` and see if that would work for us. I did this way of passing function carrying evaluations of dependencies to the function defined by `rustPlatform` as an experiment mostly. I am not certain if this could cause further breakage. Also, if this is for people to use flakes to generate a package for their config should not be really a concern as `nixpkgs` is being constantly updated by you and other maintainers as I can see now. I believe `nix develop` and `nix build` should be our focus. I would like some clarity regarding this.

Reply:
Also, just adding 
```nix
pkgs.rustPlatform.buildRustPackage {
   inherit (cargoToml.workspace.package) version;
   name = "rio";
   src = ./.;
   cargoLock.lockFile = ./Cargo.lock;
   
   cargoBuildFlags = "-p rioterm";
   
   buildInputs = runtimeDeps ++ buildDeps;
   nativeBuildInputs = buildDeps;
   runtimeDependencies = runtimeDeps;
   
   buildNoDefaultFeatures = true;
   buildFeatures = ["x11" "wayland"];
   
   meta = {
            description = "A hardware-accelerated GPU terminal emulator focusing to run in desktops and browsers";
            homepage = "https://raphamorim.io/rio";
            license = lib.licenses.mit;
            platforms = lib.platforms.unix;
            changelog = "https://github.com/raphamorim/rio/blob/master/CHANGELOG.md";
            mainProgram = "rio";
    };
};
```
was not enough to get rid of the issue where it core dumps with `NoWaylandLib` or `NoXCBLib`.

Reply:
> This seems like an old snippet from old config? What was the motivation for this? (...) Is this change necessary?

I'm not fully understanding this yet as well, but I assume that working with `pkgs` instead of an isolating function, as with `mkRio` here, potentially breaks some things, since `callPackage` does some "black magic" (that's what I got told of from the unofficial nixos-discord server).

However, [I asked on the discourse](https://discourse.nixos.org/t/correct-way-for-adding-dependencies-to-buildrustpackage/38219) what the preferred way is to declare a package and it seems to be either to move that into an extra file, [as suggested in the `nix pills`](https://nixos.org/guides/nix-pills/callpackage-design-pattern#id1426) (I only have skimmed through it at the time of writing this) or moving this into a variable, as we do here with `mkRio`.

> I believe nix develop and nix build should be our focus.

I agree, but I'd say that `nix run` is more important than `nix build` because what's the point of having a succeeding build, if you can't use it? ;)